content/tutorials/Data-channels-the-right-way-using-Ably.textile

```diff
@@ -0,0 +1,394 @@
+            var element = messageList[currentChat][index]
+            readStatus[element.user] = false
+            if (!element.user) {
+                html += '<li><small>' + element.message + '</small></li>'
+            } else {
+                html += '<li>' + element.user + ': ' + element.message + '</li>'
+            }
+        }
+        list.innerHTML = html
+        renderMembers()
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/75081aca1bcdb9af749fed982ec1b853343d9cae
+
+**What is going on in the code block above?**
+
+To chat with a user/member, we need to click the @chat now@ button next to their name. This button, in turn, calls the @chat@ method, which we have just defined.
+
+The chat method receives a single argument, i.e, the @client-id@, which is the client you want to have a chat with. We then check to be sure that the user is not trying to call himself. Although it isn’t possible for the user to call himself, we do this as an extra layer of security. At this point, if the user does not have an active connection with the other peer, it means he is the one initiating the chat, so we create a new connection with that @client_id@, passing the initiator status to be true. After this, we call the @render@ method to render the chat messages.
+
+For the two pairs to connect, the other peer has to be signaled of the incoming connection. This is where our subscription comes in. We subscribe to an event specifically for the current client’s id, which we defined the event as @ablywebrtc-signal/${clientId}@. Here, we check if the current user has a connection instance of the current client trying to connect, if not, we initiate a new connection for the other party, and surely the initiator flag is set to false.  We then call on the @handleSignal@ method of the connection class, to handle the signal coming in and connect the two pairs.
+
+Now, we can go ahead to send and receive messages.  
+
+In our @sendMessage@ method, we use the @currentChat@ variable to figure out who it is that we are trying to send a message to, then we pass our message as a string to the @send@ method of the controller class. We also push the message to the message list of the person we are chatting with. This is because WebRTC would not send the message back to the sender.
+
+Remember the @receiveMessage@ method we called in our @connection-helper.js@ file we declared earlier? Here it is. we receive the message here with the id of the user who sent it, then push it to the message list of that user. Then we call our render method.
+
+Finally, we defined the much loved @render@ method, which loops through all our messages and sets the read status to false.
+
+At this point, if we open the @index.html@ file in two separate browsers, we should be able to send and receive messages as seen below:
+
+<a href="/images/tutorials/data-channels.gif" target="_blank">
```

Comment:
Please follow the instructions in the Docs README: https://github.com/ably/docs to serve these files locally and see how they render in a browser. All the messages look quite small and hence unclear.